commit e2b67859ab6efd4458bda1baaee20331a367d995
Author: Damian Muszynski <damian.muszynski@intel.com>
Date:   Fri Feb 2 18:53:16 2024 +0800

    crypto: qat - add heartbeat error simulator
    
    Add a mechanism that allows to inject a heartbeat error for testing
    purposes.
    A new attribute `inject_error` is added to debugfs for each QAT device.
    Upon a write on this attribute, the driver will inject an error on the
    device which can then be detected by the heartbeat feature.
    Errors are breaking the device functionality thus they require a
    device reset in order to be recovered.
    
    This functionality is not compiled by default, to enable it
    CRYPTO_DEV_QAT_ERROR_INJECTION must be set.
    
    Signed-off-by: Damian Muszynski <damian.muszynski@intel.com>
    Reviewed-by: Giovanni Cabiddu <giovanni.cabiddu@intel.com>
    Reviewed-by: Lucas Segarra Fernandez <lucas.segarra.fernandez@intel.com>
    Reviewed-by: Ahsan Atta <ahsan.atta@intel.com>
    Reviewed-by: Markas Rapoportas <markas.rapoportas@intel.com>
    Signed-off-by: Mun Chun Yep <mun.chun.yep@intel.com>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/drivers/crypto/intel/qat/Kconfig b/drivers/crypto/intel/qat/Kconfig
index c120f6715a09..02fb8abe4e6e 100644
--- a/drivers/crypto/intel/qat/Kconfig
+++ b/drivers/crypto/intel/qat/Kconfig
@@ -106,3 +106,17 @@ config CRYPTO_DEV_QAT_C62XVF
 
 	  To compile this as a module, choose M here: the module
 	  will be called qat_c62xvf.
+
+config CRYPTO_DEV_QAT_ERROR_INJECTION
+	bool "Support for Intel(R) QAT Devices Heartbeat Error Injection"
+	depends on CRYPTO_DEV_QAT
+	depends on DEBUG_FS
+	help
+	  Enables a mechanism that allows to inject a heartbeat error on
+	  Intel(R) QuickAssist devices for testing purposes.
+
+	  This is intended for developer use only.
+	  If unsure, say N.
+
+	  This functionality is available via debugfs entry of the Intel(R)
+	  QuickAssist device